crates/networking/p2p/sync.rs

```diff
@@ -99,12 +108,12 @@ pub struct SyncManager {
 impl SyncManager {
     pub fn new(
         peer_table: Arc<Mutex<KademliaTable>>,
-        sync_mode: SyncMode,
+        snap_enabled: Arc<AtomicBool>,
```

Comment:
nit: can't we just pass a normal boolean here to simplify the api (even if we store it like an atomic bool)?